src/compiler/build/build-finish.ts

```diff
@@ -18,6 +18,7 @@ export const buildFinish = async (buildCtx: d.BuildCtx) => {
 };
 
 export const buildAbort = (buildCtx: d.BuildCtx) => {
+  // Why is this using the buildCtx's config and compilerCtx? This feels strange. Why are they even duplicated into the buildCtx? Ergonomics?
```

Comment:
This is the only time that the `buildCtx.compilerCtx` is used - with the exception of the usage internal to the instance of the `buildCtx`. The singleton pattern solves for this. Feels bizarre to know that these properties are duplicated into the `buildCtx`, which gets recreated over and over again throughout the course of a `build --watch` lifecycle.